hardware_interface/src/joint.cpp

```diff
@@ -0,0 +1,93 @@
+// Copyright 2020 ros2_control Development Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <string>
+#include <vector>
+
+#include "hardware_interface/component_info.hpp"
+#include "hardware_interface/joint.hpp"
+#include "hardware_interface/helpers/component_lists_management.hpp"
+#include "hardware_interface/types/hardware_interface_return_values.hpp"
+
+namespace hardware_interface
+{
+
+return_type Joint::configure(const ComponentInfo & joint_info)
+{
+  info_ = joint_info;
+  if (info_.command_interfaces.size() > 0) {
+    commands_.resize(info_.command_interfaces.size());
+  }
+  if (info_.state_interfaces.size() > 0) {
+    states_.resize(info_.state_interfaces.size());
+  }
+  return return_type::OK;
+}
+
+std::vector<std::string> Joint::get_command_interfaces() const
+{
+  return info_.command_interfaces;
+}
+
+std::vector<std::string> Joint::get_state_interfaces() const
+{
+  return info_.state_interfaces;
+}
+
+return_type Joint::get_command(
+  std::vector<double> & command, const std::vector<std::string> & interfaces) const
+{
+  return helpers::get_internal_values(command, interfaces, info_.command_interfaces, commands_);
+}
+
+void Joint::get_command(std::vector<double> & command) const
```

Comment:
why does this not have the same return value as the overload?

Reply:
There is nothing useful to return. This function should always succeed. 

Nevertheless, you are right. To be API-consistent I will add return_type::OK as return value.